# Sensor Coverage Planner Ground
sub_start_exploration_topic_ : /start_exploration
sub_terrain_map_topic_ : /terrain_map
sub_terrain_map_ext_topic_ : /terrain_map_ext
sub_state_estimation_topic_ : /state_estimation_at_scan
sub_registered_scan_topic_ : /registered_scan
sub_coverage_boundary_topic_ : /sensor_coverage_planner/coverage_boundary
sub_viewpoint_boundary_topic_ : /sensor_coverage_planner/viewpoint_boundary
sub_nogo_boundary_topic_ : /sensor_coverage_planner/nogo_boundary
pub_exploration_finish_topic_ : exploration_finish
pub_runtime_breakdown_topic_ : runtime_breakdown
pub_runtime_topic_ : /runtime
pub_waypoint_topic_ : /way_point

kAutoStart : true
kRushHome : true
kUseTerrainHeight : false
kCheckTerrainCollision : true
kExtendWayPoint : true
kUseLineOfSightLookAheadPoint : true
kNoExplorationReturnHome : true
kExtendWayPointDistance : 8.0
kKeyposeCloudDwzFilterLeafSize : 0.2
kRushHomeDist : 5
kAtHomeDistThreshold : 0.5
kTerrainCollisionThreshold : 0.5
kLookAheadDistance : 8

# PlanningEnv
kUseFrontier : true
kFrontierClusterTolerance : 1.0
kFrontierClusterMinSize : 10

kStackedCloudDwzLeafSize : 0.3
kPlannerCloudDwzLeafSize : 0.5
kCollisionCloudDwzLeafSize : 0.2
kKeyposeGraphCollisionCheckRadius : 0.4
kKeyposeGraphCollisionCheckPointNumThr : 1
kKeyposeCloudStackNum : 5
kPointCloudRowNum : 50
kPointCloudColNum : 50
kPointCloudLevelNum : 30
kMaxCellPointNum : 100000
kPointCloudCellSize : 18
kPointCloudCellHeight : 1.8
kPointCloudManagerNeighborCellNum : 5
kCoverCloudZSqueezeRatio : 2.0

# ViewPointManager
viewpoint_manager/number_x : 50
viewpoint_manager/number_y : 50
viewpoint_manager/number_z : 1
viewpoint_manager/resolution_x : 0.6
viewpoint_manager/resolution_y : 0.6
viewpoint_manager/resolution_z : 0.0
kCollisionCheckTerrainThr : 0.25
kGreedyViewPointSampleRange : 3
kLocalPathOptimizationItrMax : 10
kViewPointCollisionMargin : 0.5
kViewPointCollisionMarginZPlus : 0.2
kViewPointCollisionMarginZMinus : 0.8
kCollisionGridZScale : 1.0
kCollisionGridResolutionX : 0.2
kCollisionGridResolutionY : 0.2
kCollisionGridResolutionZ : 0.0
kCollisionPointThr : 1
kLineOfSightStopAtNearestObstacle : false
kViewPointHeightFromTerrain : 0.75
kViewPointHeightFromTerrainChangeThreshold : 0.6
kCheckDynamicObstacleCollision : false
kCollisionFrameCountMax : 3

kSensorRange : 7.5
kNeighborRange : 3.0
kCoverageOcclusionThr : 0.3
kCoverageDilationRadius : 0.5

# Grid World
kGridWorldXNum : 121
kGridWorldYNum : 121
kGridWorldZNum : 121
kGridWorldCellHeight : 3.0
kGridWorldNearbyGridNum : 5
kMinAddPointNumSmall : 40
kMinAddPointNumBig : 60
kMinAddFrontierPointNum : 20
kCellExploringToCoveredThr : 1
kCellCoveredToExploringThr: 10
kCellExploringToAlmostCoveredThr: 10
kCellAlmostCoveredToExploringThr: 20
kCellUnknownToExploringThr: 1

# Visualization
kExploringSubspaceMarkerColorGradientAlpha : true
kExploringSubspaceMarkerColorMaxAlpha : 0.8
kExploringSubspaceMarkerColorR : 0.0
kExploringSubspaceMarkerColorG : 1.0
kExploringSubspaceMarkerColorB : 0.0
kExploringSubspaceMarkerColorA : 1.0
kLocalPlanningHorizonMarkerColorR : 0.0
kLocalPlanningHorizonMarkerColorG : 1.0
kLocalPlanningHorizonMarkerColorB : 0.0
kLocalPlanningHorizonMarkerColorA : 1.0
kLocalPlanningHorizonMarkerWidth : 0.3
kLocalPlanningHorizonHeight : 3.0

# Rolling occupancy grid
rolling_occupancy_grid/resolution_x : 0.3
rolling_occupancy_grid/resolution_y : 0.3
rolling_occupancy_grid/resolution_z : 0.3